aries_cloudagent/multitenant/manager.py

```diff
@@ -222,6 +236,46 @@ async def create_wallet(
 
         return wallet_record
 
+    async def update_wallet(
+        self,
+        wallet_id: str,
+        new_settings: dict,
+    ) -> WalletRecord:
+        """Update a existing wallet and wallet record.
+
+        Args:
+            wallet_id: The wallet id of the wallet record
+            new_settings: The context settings to be updated for this wallet
+
+        Returns:
+            WalletRecord: The updated wallet record
+
+        """
+        # update wallet_record
+        async with self.profile.session() as session:
+            wallet_record = await WalletRecord.retrieve_by_id(session, wallet_id)
+
+            settings = wallet_record.settings
+            if "wallet.id" in settings:
+                del settings["wallet.id"]
+            settings.update(new_settings)
+            wallet_record.update_settings(settings)
```

Comment:
Why not just `update_settings(new_settings)`?

Reply:
Oh, it is my mistake. Has been corrected. Thanks!